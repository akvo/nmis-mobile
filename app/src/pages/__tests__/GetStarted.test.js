import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import { renderHook, act } from '@testing-library/react-native';
import GetStartedPage from '../GetStarted';

describe('GetStartedPage', () => {
  beforeAll(() => {
    const { result } = renderHook(() => useState({}));
    const [currentConfig, setCurrentConfig] = result.current;

    act(() => {
      setCurrentConfig({
        serverURL: null,
      });
    });
  });

  test('renders correctly', () => {
    const tree = renderer.create(<GetStartedPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
