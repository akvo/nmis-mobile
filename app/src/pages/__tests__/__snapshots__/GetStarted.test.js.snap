// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStartedPage renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f9fafb",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "gap": 36,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  testID="center-layout"
>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "borderRadius": 4,
        "height": 110,
        "width": 110,
      }
    }
    testID="image-skeleton"
  />
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
      }
    }
    testID="center-layout-items"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontSize": 28.285714285714285,
        }
      }
    >
      Get Started
    </Text>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontSize": 28.285714285714285,
        }
      }
    >
      collecting data the
    </Text>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontSize": 28.285714285714285,
        }
      }
    >
      smart way
    </Text>
  </View>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#242424",
      }
    }
  >
    Lorem Ipsum dolor sit amet dolor random
  </Text>
  <View
    style={
      {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        editable={true}
        onChangeText={[Function]}
        placeholder="Input Server URL"
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</View>
`;
